tests/integration/test_failed_async_inserts/test.py

```diff
@@ -0,0 +1,49 @@
+import logging
+from time import sleep
+
+import pytest
+from helpers.cluster import ClickHouseCluster
+
+
+from helpers.cluster import ClickHouseCluster
+
+cluster = ClickHouseCluster(__file__)
+node = cluster.add_instance(
+    "node", main_configs=["configs/config.xml"], with_zookeeper=True
+)
+
+
+@pytest.fixture(scope="module")
+def started_cluster():
+    try:
+        cluster.start()
+        yield cluster
+    finally:
+        cluster.shutdown()
+
+
+def test_failed_async_inserts(started_cluster):
+    node = started_cluster.instances["node"]
+
+    node.query(
+        "CREATE TABLE async_insert_30_10_2022 (id UInt32, s String) ENGINE = Memory"
+    )
+
+    try:
+        node.query("SET async_insert = 1")
+        node.query("INSERT INTO async_insert_30_10_2022 VALUES ()")
+        node.query("INSERT INTO async_insert_30_10_2022 VALUES ([1,2,3], 1)")
+        node.query(
+            'INSERT INTO async_insert_30_10_2022 format JSONEachRow {"id" : 1} {"x"}'
+        )
+        node.query("INSERT INTO async_insert_30_10_2022 Values (throwIf(4),'')")
+    except Exception:
+        return None
```

Comment:
That won't work. First, `async_insert` will be enabled only for query `SET async_insert = 1`, other queries will be executed withot enabled `async_insert`. Second, the second query will throw an exception, we cathc ot in `except` and return from the test without testing anything. Let's rewrite it a bit:
```suggestion
        node.query(
        "INSERT INTO async_insert_30_10_2022 SETTINGS async_insert = 1 VALUES ()",
        ignore_error=True,
    )
    node.query(
        "INSERT INTO async_insert_30_10_2022 SETTINGS async_insert = 1 VALUES ([1,2,3], 1)",
        ignore_error=True,
    )
    node.query(
        'INSERT INTO async_insert_30_10_2022 SETTINGS async_insert = 1 FORMAT JSONEachRow {"id" : 1} {"x"}',
        ignore_error=True,
    )
    node.query(
        "INSERT INTO async_insert_30_10_2022 SETTINGS async_insert = 1 VALUES (throwIf(4),'')",
        ignore_error=True,
    )
```